### PR TITLE
Fix: fix configuration not found health check error

### PR DESCRIPTION
**Pull request by FogDong**
Signed-off-by: FogDong <dongtianxin.tx@alibaba-inc.com>

### Description of your changes

fix configuration not found health check error

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/controller/core.oam.dev/v1alpha2/application/apply.go [hunk 1]
```diff
@@ -231,7 +232,8 @@ func (h *AppHandler) collectHealthStatus(ctx context.Context, wl *appfile.Worklo
 	if wl.CapabilityCategory == types.TerraformCategory {
 		var configuration terraformapi.Configuration
 		if err := h.r.Client.Get(ctx, client.ObjectKey{Name: wl.Name, Namespace: namespace}, &configuration); err != nil {
-			return nil, false, errors.WithMessagef(err, "app=%s, comp=%s, check health error", appName, wl.Name)
+			klog.ErrorS(err, "failed to get configuration", "app name", appName, "comp name", wl.Name)
+			return nil, false, nil
```
Comment (wonderflow) [inline pkg/controller/core.oam.dev/v1alpha2/application/apply.go:236]: why don't return the error here?

Comment (anoop2811) [reply pkg/controller/core.oam.dev/v1alpha2/application/apply.go:236]: Same question as @wonderflow . Maybe I am missing what the fix here is as it only seems to log the error.

Comment (FogDong) [reply pkg/controller/core.oam.dev/v1alpha2/application/apply.go:236]: If we return the error here, in the application status, we can see the `configuration not found error...`. but this error is caused by we failed to get configuration when we check the health.

Comment (wonderflow) [reply pkg/controller/core.oam.dev/v1alpha2/application/apply.go:236]: the root cause is why we will miss the configuration here? Will we periodically check the existence of configuration?

**Pull request closed without merging**